MapboxNavigation/NavigationComponent.swift

```diff
@@ -5,43 +5,43 @@ import CoreLocation
 /*
  A navigation component is a member of the navigation UI view hierarchy that responds as the user progresses along a route according to the `NavigationServiceDelegate` protocol.
  */
-@objc public protocol NavigationComponent: NavigationServiceDelegate {}
+public typealias NavigationComponent = NavigationServiceDelegate
```

Comment:
Do we expect that NavigationComponent will never have method and property requirements of its own beyond the protocols that the type conforms to?

This means new navigation component types no longer need to explicitly declare that they’re navigation components. Instead, a type becomes a navigation component automatically by virtue of conforming to NavigationServiceDelegate. Could that lead to a class being used in situations it didn’t expect to be used in?